isis/src/system/apps/isisdataeval/tools/README.md

```diff
@@ -0,0 +1,115 @@
+
+optional arguments:
+  -h, --help           show this help message and exit
+  --isisdata ISISDATA  ISISDATA directory root
+  --outpath OUTPATH    NEW ISISDATA directory to create
+  --ghostdir GHOSTDIR  Replaces the --isisdata path with this string to ghost the actual input
+                       directory path
+  --saveconfig, -s     Retain *.db, *.conf files rather than replace with processs info
+  --dryrun, -n         Only print actions but do not execute
+  --verbose, -v        Verbose output
+
+isisdata_mockup.py will create a copy of all the files found in the directory specified by
+the --isisdata parameter. All files and directories contained in --isisdata will
+be copied to the --outpath destination, which specifies a new directory. This
+directory will be created if it does not exist.
+
+All *.db and *.conf files are copied as is, fully intact. Also needed to use this
+mock up of ISISDATA is the NAIF LSK. This LSK kernel is always loaded by the iTime
+class to concert UTC and Et times. Otherwise, all other files encountered will
+be created but the contents are replaced by information regarding the --isisdata
+source file. This information includes the original file opath, creation and
+modification dates in UTC format, the size of the file and its hash values.
+
+This app sets up a full data directory structure that mimics the contents of
+ISISDATA (or any directory for that matter). The size of the file, its creation
+and last modification dates and the md5, sha1 and sha256 hash values are created
+in a dict which is then written to the new output file if its not a special ISIS
+kernel db/conf file or LSK.
+
+Finally, by setting --ghostdir '$ISISDATA', this now provides a connection to
+the source file in the ISISDATA directory. This can be used to compare hash values
+compute by other sources.
+
+Example:
+
+To provide a full mock up of ISISDATA directory:
+
+isisdata_mockup.py --saveconfig --isisdata /opt/isis/data --outpath $PWD/mockisisdata --ghostdir '$ISISDATA'
+
+Author: Kris J. Becker, University of Arizona
+        kbecker@orex.lpl.arizona.edu
+
+History 2023-03-15 Kris Becker - Original verison
+```
+
+You can also choose to process only a single directory, as shown in the following example:
+
+```
+isisdata_mockup.py --saveconfig --isisdata /opt/isis/data/voyager1  --outpath $PWD/mockisisdata/voyager1 --ghostdir '$ISISDATA/voyager1'
+```
+Processing times can be significant since this script is computing 3 different hash values per file.
+
+
+# <a name="#isisdatamockupinstall">ISISDATA Mockup Test Data Preparation</a>
+
+With this tool and the files listed in `isisdataeval_isisdata_mockup_files.lis`, the test for isisdataeval can be recreated from any ISISDATA installation. Note that from time to time, the files used in this ISISDATA test could change which would cause failures. Here are the commands to create the `isisdataeval` test ISISDATA mockup directory - from the Git install directory:
+
```

Comment:
Appreciate the thorough instructions.  Is it possible to simple make this mocked ISISDATA area static and not rely on the real data for comparison? From the tests it seems it's comparing hashes, my first thought is that we don't need to compare hashes from the original file? probably something I am missing.

Reply:
Sure its possible to make data up, but as I mentioned above, this does not validate $ISISDATA. That would tell us nothing about ISISDATA or `isisdataeval` unless some additional effort or procedures are pursued on real data. These tests use selected files from four very stable $ISISDATA directories that should not change. This approach integrates verification and validation of required data for proper and more thorough testing of the ISIS system. 

Secondarily, but just as important, this provides tools to that can help address ongoing problems with ISISDATA encountered by ISIS users by first ensuring the problem is not related to their local ISISDATA installation.

Reply:
I agree on the second part, after this is merged I plan to add it to the nightly ISISDATA script and find a good way to report results in AWS such that we can get a better idea of the changes other than did download/did not download. 

I think there might be a way to test the hash comparisons without real data, as the goal here is to test the app not the data itself, the data itself would be tested elsewhere. But there is already a ton of good work here and I'd rather get it in than split hairs on testing.

Reply:
Great. Can you hold off for a bit on the merge.I have an update coming.

It might address some of the things you mentioned.